evm_arithmetization/src/recursive_verifier.rs

```diff
@@ -427,6 +427,21 @@ pub(crate) fn get_memory_extra_looking_sum_circuit<F: RichField + Extendable<D>,
         );
     });
 
+    #[cfg(feature = "cdk_erigon")]
+    let burn_addr = match public_values.burn_addr {
+        BurnAddrTarget::BurnAddr(addr) => addr,
+        BurnAddrTarget::Burnt() => panic!("There should be an address set in cdk_erigon."),
+    };
+    #[cfg(feature = "cdk_erigon")]
+    let mut sum = add_data_write(
+        builder,
+        challenge,
+        sum,
+        metadata_segment,
+        GlobalMetadata::BurnAddr.unscale(),
+        &burn_addr,
+    );
```

Comment:
The problem with this is that `sum` then becomes a local variable (since it is within a block). Clippy then complains that the value does not need to be mutable (and if I remove the `mut`, it complains that the value is unused). Also, given those errors, I assume that means the recursive verification would also fail, since I doubt the sum would have actually been updated.

Reply:
That's because you're _initializing_ the `sum` conditionally, but you don't need to as the variable already exists (I just copy pasted your code and adapted it. If you remove the `let mut` it works fine:

```rust
    #[cfg(feature = "cdk_erigon")]
    {
        let burn_addr = match public_values.burn_addr {
            BurnAddrTarget::BurnAddr(addr) => addr,
            BurnAddrTarget::Burnt() => panic!("There should be an address set in cdk_erigon."),
        };
        sum = add_data_write(
            builder,
            challenge,
            sum,
            metadata_segment,
            GlobalMetadata::BurnAddr.unscale(),
            &burn_addr,
        );
    }
 ```

Reply:
Indeed... I thought I had done that and there was another issue for some reason...